lib/filterx/expr-null-coalesce.c

```diff
@@ -59,26 +59,36 @@ _eval(FilterXExpr *s)
   return lhs_object;
 }
 
-FilterXExpr *
-filterx_null_coalesce_new(FilterXExpr *lhs, FilterXExpr *rhs)
+static FilterXExpr *
+_optimize(FilterXExpr *s)
 {
-  if (filterx_expr_is_literal(lhs))
+  FilterXNullCoalesce *self = (FilterXNullCoalesce *) s;
+
+  if (filterx_binary_op_optimize_method(s))
+    g_assert_not_reached();
+
+  if (filterx_expr_is_literal(self->super.lhs))
```

Comment:
[Nitpick, absolutely optional]

Thanks for retaining the original structure while introducing the `optimize` method. Would you be willing to refactor this to use early returns where possible?

Reply:
Not a problem, thanks for the reminder! Done :)